src/Quadrature.jl

```diff
@@ -403,22 +403,35 @@ ZygoteRules.@adjoint function __solvebp(prob,alg,sensealg,lb,ub,p,args...;kwargs
             _f = (x) -> prob.f(dx,x,p)
             if sensealg.vjp isa ZygoteVJP
                 dfdp = function (dx,x,p)
-                    _,back = Zygote.pullback(p) do p
-                        _dx = Zygote.Buffer(x, prob.nout, size(x,2))
-                        prob.f(_dx,x,p)
-                        copy(_dx)
+                    for idx in 1:size(x,2)
+                        _,back = Zygote.pullback(p) do p
+                            _dx = Zygote.Buffer(x, prob.nout)
+                            prob.f(_dx,@view(x[:,idx]),p)
+                            copy(_dx)
+                        end
+                        dx[:,idx] .= back(y)[1]
                     end
-                    back(y)[1]
                 end
             elseif sensealg.vjp isa ReverseDiffVJP
                 error("TODO")
             end
         else
             _f = (x) -> prob.f(x,p)
             if sensealg.vjp isa ZygoteVJP
-                dfdp = function (x,p)
-                    _,back = Zygote.pullback(p->prob.f(x,p),p)
-                    back(y)[1]
+                if prob.batch > 0
+                    dfdp = function (x,p)
+                        out = zeros(length(p),size(x,2))
+                        for idx in 1:size(x,2)
```

Comment:
I guess that gets it working, but there should be a better way to batch this?

Reply:
Yeah, I saw this a low hanging fruit. I'm all ears on thoughts for a better approach.

Reply:
how is this not equivalent to the other one?